### PR TITLE
feat(structured-list): add experimental styles and layout

### LINKED ISSUE #1275: Experimental Component - Structured List - Development
##### Development
A developer can discover and install the experimental component by following accessible documentation.

- [x] Add to carbon-components
- [ ] Update website (add to the experimental section and update component status page)

### PR DESCRIPTION
**Pull request by aledavila**
Closes #1275 

#### Changelog

**New**

**Changed**

- background color
- font size and color
- position of the selectable structure list checkbox

#### Testing / Reviewing

check dev environment is not broken

Staging Link: 
http://structured-list-x-appreciative-gelada.mybluemix.net/?nav=structured-list

### CODE REVIEW

**Review by asudoh — COMMENTED** [review]
LGTM basically, my only question is if we have a plan to create color tokens for new colors introduced here. Thanks @aledavila!

File: src/components/structured-list/_structured-list.scss [hunk 1]
```diff
@@ -174,3 +174,157 @@
...
+      background-color: $ibm-colors__gray--20;
+    }
+
+    &.#{$prefix}--structured-list-row--header-row {
+      border-bottom: 1px solid $ibm-colors__gray--20;
+      cursor: inherit;
+    }
+
+    &:focus:not(.#{$prefix}--structured-list-row--header-row) {
+      outline: 2px solid $ibm-colors__blue--60;
```
Comment (alisonjoseph) [inline src/components/structured-list/_structured-list.scss]: Should be able to use the focus mixin here, looks like the spec has it on the inside anyway. 
```suggestion
      @include focus-outline('outline');
```

File: src/components/structured-list/_structured-list.scss [hunk 2]
```diff
@@ -174,3 +174,157 @@
...
+    }
+  }
+
+  .#{$prefix}--structured-list-row {
+    display: table-row;
+    border-bottom: 1px solid $ui-03;
+    transition: $transition--base $carbon--standard-easing;
+
+    .#{$prefix}--structured-list--selection &:hover:not(.#{$prefix}--structured-list-row--header-row) {
+      background-color: $ibm-colors__gray--10;
```
Comment (alisonjoseph) [inline src/components/structured-list/_structured-list.scss]:
```suggestion
      background-color: $ui-01;
```

File: src/components/structured-list/_structured-list.scss [hunk 3]
```diff
@@ -174,3 +174,157 @@
...
+      cursor: pointer;
+    }
+
+    &.#{$prefix}--structured-list-row--selected {
+      background-color: $ibm-colors__gray--20;
```
Comment (alisonjoseph) [inline src/components/structured-list/_structured-list.scss]:
```suggestion
      background-color: $ui-03;
```

File: src/components/structured-list/_structured-list.scss [hunk 4]
```diff
@@ -174,3 +174,157 @@
...
+    }
+
+    &.#{$prefix}--structured-list-row--header-row {
+      border-bottom: 1px solid $ibm-colors__gray--20;
```
Comment (alisonjoseph) [inline src/components/structured-list/_structured-list.scss]:
```suggestion
      border-bottom: 1px solid $ui-03;
```

File: src/components/structured-list/_structured-list.scss [hunk 5]
```diff
@@ -174,3 +174,157 @@
...
+    @include reset;
+    @include typescale('zeta');
+    @include line-height('body');
+    @include padding-td--x;
+    position: relative;
+    display: table-cell;
+  }
+
+  .#{$prefix}--structured-list-th {
+    color: $ibm-colors__gray--100;
```
Comment (alisonjoseph) [inline src/components/structured-list/_structured-list.scss]:
```suggestion
    color: $text-01;
```

File: src/components/structured-list/_structured-list.scss [hunk 6]
```diff
@@ -174,3 +174,157 @@
...
+  }
+  .#{$prefix}--structured-list-td {
+    color: $ibm-colors__gray--70;
```
Comment (alisonjoseph) [inline src/components/structured-list/_structured-list.scss]:
```suggestion
    color: $text-02;
```

File: src/components/structured-list/_structured-list.scss [hunk 7]
```diff
@@ -174,3 +174,157 @@
...
+    vertical-align: middle;
+    transition: $transition--base $carbon--standard-easing;
+
+    .#{$prefix}--structured-list-row:hover & {
+      fill: $ibm-colors__gray--40;
+    }
+
+    .#{$prefix}--structured-list-input:checked + .#{$prefix}--structured-list-row &,
+    .#{$prefix}--structured-list-input:checked + .#{$prefix}--structured-list-td & {
+      fill: $ibm-colors__gray--100;
```
Comment (alisonjoseph) [inline src/components/structured-list/_structured-list.scss]:
```suggestion
      fill: $text-01;
```

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
LGTM other than adding the focus mixin and color tokens. 👍 

Although can you explain why an html change is needed? We've been able to avoid them on other components up until this point (other than tags, but that's just class names). On the surface looking at the two designs its hard to tell since they seem to be very similar. 

Oh and don't forget to name the PR so that it will trigger semantic release

**Comment by aledavila** [comment]
@alisonjoseph it was because the check selected check is now at the end instead of the of the first column. I couldn't do a row reverse because it would put everything out of order. This is the order:

v9:
- input (triggers the svg check)
- svg check
- columns

v10: 
- columns
- input
- svg check

**Comment by alisonjoseph** [comment]
Ah ok 👍 that makes sense then. 

Can you make an issue for React for this component and tags? We will need to support the updated html since we can't just swap the feature flag in sass anymore.

**Comment by aledavila** [comment]
Yup can do

**Comment by laurenmrice** [comment]
One thing I noticed
Selection and Hover: Text in selected and hover rows needs to be Gray 100 instead of Gray 70 :) otherwise looks good to me!

**Comment by aledavila** [comment]
@alisonjoseph & @laurenmrice it is ready for review with all the updates. thanks!

**Comment by laurenmrice** [comment]
Couple things:

-When a row is on hover, the line rules on the bottom and the top of the hover states should not appear.
screen shot 2018-10-30 at 1 03 34 pm

-If a row has already been selected, there should not be a hover state on top of the selected state.

-On Focus state, the text can remain Gray 70 instead of switching to Gray 100.

File: src/components/structured-list/_structured-list.scss [hunk 8]
```diff
@@ -174,3 +174,179 @@
...
+
+    // &:after {
+    //   position: absolute;
+    //   content: '';
+    //   // display: block;
+    //   width: 100%;
+    //   height: rem(1px);
+    //   background-color: $ui-01;
+    //   top: -1px;
+    // }
```
Comment (tw15egan) [inline src/components/structured-list/_structured-list.scss]: ☠️ Delete me ☠️

**Comment by alisonjoseph** [comment]
Not sure if this was ready to re-review or not @aledavila but seeing a border/outline flash on hover

**Comment by aledavila** [comment]
@alisonjoseph  @laurenmrice yes its ready now.

**Comment by laurenmrice** [comment]
Looks good. 👌

**Review by tw15egan — APPROVED** [review]
Looks great to me! 👍 🎉

**Review by alisonjoseph — APPROVED** [review]
looks great!

**Pull request merged**